Documentation/test-suites-overview.md

```diff
@@ -0,0 +1,240 @@
+be found at [mender-qa
+repository](https://github.com/mendersoftware/mender-qa/blob/master/scripts/yocto-build-and-test.sh).
+
+The following recipe gives an example of how to build and test the
+`qemux86-64-uefi-grub` platform.
+
+```
+# Initialize a new Yocto build environment
+source oe-init-build-env build-qemux86-64-uefi-grub
+# Add meta-mender build layers
+bitbake-layers add-layer ../meta-mender/meta-mender-core
+bitbake-layers add-layer ../meta-mender/meta-mender-demo
+bitbake-layers add-layer ../meta-mender/meta-mender-qemu
+# Add meta-mender test layer
+bitbake-layers add-layer ../meta-mender/tests/meta-mender-ci
+# Take the default Yocto configuration for this platform
+cp ../meta-mender/tests/build-conf/qemux86-64-uefi-grub/local.conf conf/
+# Set MENDER_ARTIFACT_NAME
+cat >> conf/local.conf <<EOF
+MENDER_ARTIFACT_NAME = "mender-image-local"
+EOF
+# Build full image
+bitbake core-image-full-cmdline
+# Run the tests!
+cd ../meta-mender/tests/acceptance/
+python3 -m pytest -p no:xdist --bitbake-image core-image-full-cmdline --board-type=qemux86-64
+```
+
+## Backend tests
+
+Test suite       | backend integration tests
+---------------- | ----------
+ISTQB equivalent | System integration testing and server-only Acceptance testing
+Test object      | The Mender backend server, as a whole
+Source code      | [integration/backend-tests](https://github.com/mendersoftware/integration/tree/master/backend-tests)
+How to run       | [`./run`](https://github.com/mendersoftware/integration/blob/master/backend-tests/run)
+Dependencies     | Docker, docker-compose
+
+This test suite verifies the interfaces and behavior of a stand-alone Mender
+server installation as a whole. A real client is not used, and instead API calls
+to the different endpoints simulates Mender clients connecting to the Mender
+server.
+
+Refer to the
+[README](https://github.com/mendersoftware/integration/tree/master/backend-tests)
+for more details of this test suite.
+
+
+## Integration tests
+
+Test suite       | integration tests
+---------------- | ----------
+ISTQB equivalent | System testing and Acceptance testing
+Test object      | The Mender product, as a whole
+Source code      | [integration/tests](https://github.com/mendersoftware/integration/tree/master/tests)
+How to run       | [`./run.sh`](https://github.com/mendersoftware/integration/blob/master/tests/run.sh)
+Dependencies     | Docker, docker-compose, Python 3, [Python packages](https://github.com/mendersoftware/integration/blob/master/tests/requirements/python-requirements.txt).
+
+This is the highest level of testing we have. The integration test suite
+verifies Mender as a product, including both the server and the client.
+Different scenarios and configurations are taken into consideration covering
+both functional and non-functional testing.
+
+Refer to the
+[README](https://github.com/mendersoftware/integration/tree/master/tests) for
+more details of this test suite.
+
```

Comment:
```suggestion
## E2E tests

Test suite       | e2e tests
---------------- | ----------
ISTQB equivalent | System testing and Acceptance testing
Test object      | The Mender product, as a whole
Source code      | [gui/tests/e2e_tests](https://github.com/mendersoftware/gui/tree/master/tests/e2e_tests)
How to run       | [see e2e-test stage](https://github.com/mendersoftware/gui/blob/master/.gitlab-ci.yml)
Dependencies     | Docker, docker-compose, [mender-demo-artifact](https://dgsbl4vditpls.cloudfront.net/mender-demo-artifact.mender).

This is the highest level of testing we have. The integration test suite
verifies Mender as a product, including both the server and a client.
Different scenarios and configurations are taken into consideration covering
both functional and non-functional testing. This also takes the Mender SaaS
offering into account with appropriate tests run depending on the environment.

Refer to the
[README](https://github.com/mendersoftware/integration/tree/master/tests) for
more details of this test suite.

### Running locally

The e2e tests are run in a separate container within the same network as the
service under test and other services when required. The GUI repository has a
Docker compose file (usually named `docker-compose-acceptance.yml`) describing
other services, networks, and aliases that are required to run the tests.

An example of running the OS tests from the `GUI` follows:

´´´
docker build -t mendersoftware/gui:pr .
wget -qP tests/e2e_tests/cypress/fixtures "https://dgsbl4vditpls.cloudfront.net/mender-demo-artifact.mender"
git clone --single-branch https://github.com/mendersoftware/integration.git
GUI_REPOSITORY=$(pwd) INTEGRATION_PATH=$(pwd)/integration ./tests/e2e_tests/run
´´´
```

Reply:
:clap: Thanks!

Reply:
I modified a bit the integration tests text, and this one I wrote:

```
The End to End tests (E2E) verify also Mender as a product, including the GUI.
The client interactions are simulated with scripts, while the backend
interactions are carried on through the GUI, the same way a human would do it
when using the product.
```